pkg/kubelet/config/config_test.go

```diff
@@ -189,8 +189,8 @@ func TestInvalidPodFiltered(t *testing.T) {
 	channel <- podUpdate
 	expectPodUpdate(t, ch, CreatePodUpdate(kubetypes.ADD, TestSource, CreateValidPod("foo", "new")))
 
-	// add an invalid update
-	podUpdate = CreatePodUpdate(kubetypes.UPDATE, TestSource, &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}})
+	// add an invalid update, pod with same name
+	podUpdate = CreatePodUpdate(kubetypes.ADD, TestSource, CreateValidPod("foo", "new"))
 	channel <- podUpdate
 	expectNoPodUpdate(t, ch)
```

Comment:
just random thought, feel free to take it or not. Should we ensure `podStorage.Merge` has completed before we checking channel `ch` or at least wait for a while in `expectNoPodUpdate`? Otherwise we might receive event from `ch` after a while but `expectNoPodUpdate` fails to notice that.

Reply:
expectNoPodUpdate is used inside of expectPodUpdate, so it would delay all the tests using that function, and I did not want to slow everything down for a probabilistic check. Do you have a suggestion for how I can force it?

Reply:
One thing comes to mind is adding and removing a pod expecting corresponding updates in `ch`. If something unexpected was delayed on the channel from the previous update - expected updates on this additional add/remove combination wouldn't match.

But this may be a separate work item I think.

Reply:
/lgtm